tools/reviewer/job/impl/SubmitterTask.java

```diff
@@ -0,0 +1,187 @@
+        initializeFirestoreClientIfNull();
+        // noinspection unchecked
+        List<Diff> diffs = (List) this.firestoreClient.listDocuments(DIFF_PATH, Diff.newBuilder());
+
+        if (!fileUtils.folderExists("submitter")) {
+          fileUtils.mkdirs("submitter");
+        }
+
+        diffs
+            .stream()
+            // get Diffs in SUBMITTING state
+            .filter(diff -> diff.getStatus().equals(Diff.Status.SUBMITTING))
+            // for each of them, get latest CIResponse (get or pop?)
+            .map(
+                diff -> {
+                  MessageWithId responseWithId =
+                      this.firestoreClient.getDocumentFromCollection(
+                          String.format(CI_RESPONSES_HISTORY_PATH, diff.getId()),
+                          CIResponse.newBuilder());
+                  CIResponse response = (CIResponse) responseWithId.message();
+                  return Pair.of(diff, response);
+                })
+            // only proceed if CIResponse was present
+            .filter(diffCiPair -> diffCiPair.getValue() != null)
+            // only proceed if TargetResult.success = true for all Results
+            .filter(
+                diffCiPair ->
+                    diffCiPair
+                        .getValue()
+                        .getResultsList()
+                        .stream()
+                        .allMatch(CIResponse.TargetResult::getSuccess))
+            .forEach(
+                diffCiPair -> {
+                  Diff diff = diffCiPair.getKey();
+                  CIResponse response = diffCiPair.getValue();
+
+                  boolean shouldPush = true;
+                  List<GitRepo> gitRepos = new ArrayList<>();
+
+                  CIRequest.Builder newRequestBuilder =
+                      CIRequest.newBuilder().setForSubmission(true);
+
+                  for (CIResponse.TargetResult targetResult : response.getResultsList()) {
+                    CIRequest.Target target = targetResult.getTarget();
+
+                    Repo repo = target.getRepo();
+                    String repoPath =
+                        fileUtils.joinPaths(
+                            fileUtils.getCurrentWorkingDirectory(), "submitter", repo.getId());
+                    GitRepo gitRepo = this.gitRepoFactory.create(repoPath);
+                    gitRepos.add(gitRepo);
+
+                    try {
+                      if (fileUtils.folderEmptyOrNotExists(repoPath)) {
+                        gitRepo.cloneRepo(repo.getUrl(), repoPath);
+                      } else {
+                        gitRepo.pull();
+                      }
+                    } catch (IOException e) {
+                      e.printStackTrace();
+                    }
+
+                    String branch = String.format("origin/D%d", diff.getId());
+                    gitRepo.switchBranch(branch);
+
+                    newRequestBuilder =
+                        newRequestBuilder.addTarget(
```

Comment:
Do we need to assign this back to `newRequestBuilder`?

Reply:
@oferb we probably don't as it compiles to `this = this` (`this` being `newRequestBuilder`) which doesn't make much sense

Reply:
@oferb ae17bcd6192338e64122e84142d2a2c4c2d42604